libraries/rush-lib/src/utilities/npmrcUtilities.ts

```diff
@@ -66,11 +46,30 @@ function _trimNpmrcFile(options: {
       const environmentVariables: string[] | null = line.match(expansionRegExp);
```

Comment:
Consider replacing `expansionRegExp` with `(?:\$\{)([^\}:-]+)(?:\:?-(.+))?(?:\})`. Then `match[0]` is the env var name and `match[1]` is the fallback value.

Reply:
I found that when encountering multi-environment variables like `var1=${foo}-${bar}`, this regular expression fails.